crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -683,4 +689,27 @@ mod test {
         stage.run(&mut world);
         assert!(*world.get_resource::<bool>().unwrap(), "after test");
     }
+
+    #[test]
+    fn repeated_on_in_stack_triggers() {
+        let mut world = World::default();
+
+        world.insert_resource(0u32);
+        world.insert_resource(State::new(MyState::S1));
+
+        let mut stage = SystemStage::parallel();
+
+        stage.add_system_set(State::<MyState>::get_driver());
+
+        stage.add_system_set(SystemSet::on_in_stack_update(MyState::S1).with_system(
+            |mut counter: ResMut<u32>, mut state: ResMut<State<MyState>>| {
+                *counter += 1;
+                state.push(MyState::S2).ok();
+            },
+        ));
+        stage.run(&mut world);
+        stage.run(&mut world);
+        // This is 3 even though we run twice, because the first actually does two so MyState::S2 is also updated.
```

Comment:
This comment is unclear. What does "because the first actually does two" mean?

Reply:
To quote an earlier comment
> The reason first run does twice is because the first does an "update" twice, once with S1 as the active state, and then another with S2 as the active state. When updating S2, systems for S1's on_in_stack also need to run.

Should I put this there as well?

Reply:
Ah, I see. Eh it's fine.